File: FigurasGeometricas/FigurasGeometricas.py
from multimethod import multimethod
import math

class FigurasGeometricas:
    # Círculo
    @multimethod
    def area(self, radio: float) -> float:
        return math.pi * radio * radio

    # Rectángulo
    @multimethod
    def area(self, base: float, altura: float) -> float:
        return base * altura

    # Triángulo rectángulo
    @multimethod
    def area(self, base: int, altura: int) -> float:
        return 0.5 * base * altura

    # Trapecio
    @multimethod
    def area(self, baseMayor: float, baseMenor: float, altura1: float) -> float:
        return ((baseMayor + baseMenor) * altura1) / 2

    # Pentágono regular
    @multimethod
    def area(self, lado: int, apotema: float) -> float:
        return (5 * lado * apotema) / 2

fig = FigurasGeometricas()
print("Área del círculo:", fig.area(5))
print("Área del rectángulo:", fig.area(4.0, 6.0))
print("Área del triángulo rectángulo:", fig.area(3, 4))
print("Área del trapecio:", fig.area(6.0, 4.0, 5.0))
print("Área del pentágono:", fig.area(4, 5.0))

File: FigurasGeometricas/FigurasGeometricas.java
class FigurasGeometricas  {
    // Círculo
    public double area(double radio) {
        return Math.PI * radio * radio;
    }
    // Rectángulo
    public double area(double base, double altura) {
        return base * altura;
    }
    // Triángulo rectángulo
    public double area(int base, int altura) {
        return 0.5 * base * altura;
    }
    // Trapecio
    public double area(double baseMayor, double baseMenor, double altura1) {
        return ((baseMayor + baseMenor) * altura1) / 2;
    }
    // Pentágono regular
    public double area(int  lado, double  apotema) {
        return (5 * lado * apotema) / 2;
    }

    public static void main(String[] args) {
        FigurasGeometricas fig = new FigurasGeometricas();
        System.out.println("Área del círculo: " + fig.area(5));
        System.out.println("Área del rectángulo: " + fig.area(4, 6));
        System.out.println("Área del triángulo rectángulo: " + fig.area(3, 4));
        System.out.println("Área del trapecio: " + fig.area(6, 4, 5));
        System.out.println("Área del pentágono: " + fig.area(4, 5));
    }
}
